src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,169 @@
+
 package core.basesyntax;
 
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T data) {
+            this.data = data;
+        }
+    }
+
+    private boolean checkIndex(int index) {
+        if (index >= 0 && index < size) {
+            return true;
+        } else {
+            throw new IndexOutOfBoundsException("Wrong index" + index);
+        }
+    }
+
+    private Node<T> findIndex(int index) {
+        Node<T> current = first;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
+    }
+
+    private void removeLogic() {
+        Node<T> current = first;
+        if (current == first) {
+            first = current.next;
+        } else {
+            current.prev.next = current.next;
+        }
+        if (current == last) {
+            last = current.prev;
+        } else {
+            current.next.prev = current.prev;
+        }
+        size--;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            if (isEmpty()) {
+                last = newNode;
+            } else {
+                first.prev = newNode;
+                newNode.next = first;
+                first = newNode;
+                size++;
+            }
+        } else {
+            checkIndex(index);
+            Node<T> current = findIndex(index);
+            newNode.next = current;
+            newNode.prev = current.prev;
+            current.prev.next = newNode;
+            current.prev = newNode;
+            size++;
+        }
```

Comment:
```suggestion
        Node<T> newNode = new Node<>(value);
        if (index == size) {
            add(value);
            return;
        }
        if (index == 0) {
            first.prev = newNode;
            newNode.next = first;
            first = newNode;
        } else {
            checkIndex(index);
            Node<T> current = findIndex(index);
            newNode.next = current;
            newNode.prev = current.prev;
            current.prev.next = newNode;
            current.prev = newNode;
        }
        size++;
```

we could simplify this